recipes/zeromq/all/conanfile.py

```diff
@@ -0,0 +1,111 @@
+import os
+from conans import ConanFile, tools, CMake
+
+
+class ZeroMQConan(ConanFile):
+    name = "zeromq"
+    homepage = "https://github.com/zeromq/libzmq"
+    description = "ZeroMQ is a community of projects focused on decentralized messaging and computing"
+    topics = ("conan", "zmq", "libzmq", "message-queue", "asynchronous")
+    url = "https://github.com/conan-io/conan-center-index"
+    license = "LGPL-3.0"
+    exports_sources = ["CMakeLists.txt"]
+    settings = "os", "arch", "compiler", "build_type"
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False],
+        "encryption": [None, "libsodium", "tweetnacl"],
+    }
+    default_options = {
+        "shared": False,
+        "fPIC": True,
+        "encryption": "libsodium",
+    }
+    generators = "cmake", "cmake_find_package"
+
+    _cmake = None
+    _source_subfolder = "source_subfolder"
+    _build_subfolder = "build_subfolder"
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def configure(self):
+        if self.options.shared:
+            del self.options.fPIC
+
+    def requirements(self):
+        if self.options.encryption == "libsodium":
+            self.requires.add("libsodium/1.0.18")
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        os.rename("libzmq-{}".format(self.version), self._source_subfolder)
+
+    def _configure_cmake(self):
+        if self._cmake:
+            return self._cmake
+        self._cmake = CMake(self)
+        self._cmake.definitions["ENABLE_CURVE"] = self.options.encryption is not None
+        self._cmake.definitions["WITH_LIBSODIUM"] = self.options.encryption == "libsodium"
+        self._cmake.definitions["ZMQ_BUILD_TESTS"] = False
+        self._cmake.definitions["WITH_PERF_TOOL"] = False
+        self._cmake.definitions["BUILD_SHARED"] = self.options.shared
+        self._cmake.definitions["BUILD_STATIC"] = not self.options.shared
+        self._cmake.definitions["ENABLE_CPACK"] = False
+        self._cmake.configure(build_folder=self._build_subfolder)
+        return self._cmake
+
+    def _patch_sources(self):
+        os.unlink(os.path.join(self._source_subfolder, "builds", "cmake", "Modules", "FindSodium.cmake"))
+        os.rename("Findlibsodium.cmake", "FindSodium.cmake")
+        tools.replace_path_in_file(os.path.join(self._source_subfolder, "CMakeLists.txt"),
```

Comment:
@uilianries 
I just tried building zeromq locally and conan is stuck on this line:
```
tools.replace_path_in_file(os.path.join(self._source_subfolder, "CMakeLists.txt"),
                                   "SODIUM_FOUND",
                                   "libsodium_FOUND")
```

Reply:
file permission maybe? I can't remember a similar situation when running `replace_path_in_file` on Linux.

Reply:
I was using `tools.replace_path_in_file` instead of `tools.replace_in_file` ... 
 :no_mouth:

Reply:
You might protect your CI services because this bug caused the `conan create` to take forever and slowly eat memory.

Reply:
Thanks for the hint @madebr! I also noticed that behavior but couldn't tell where was the issue. I will open an issue about it. Thanks!